src/Umbraco.Web/UmbracoHelper.cs

```diff
@@ -70,24 +70,31 @@ internal UmbracoHelper()
         #endregion
 
         // ensures that we can return the specified value
+        [Obsolete("This method is only used in Obsolete properties")]
         T Ensure<T>(T o) where T : class => o ?? throw new InvalidOperationException("This UmbracoHelper instance has not been initialized.");
 
+        [Obsolete("Inject and use an instance of " + nameof(IUmbracoComponentRenderer) + " in the constructor for using it in classes or get it from Current.UmbracoComponentRenderer in views.")]
         private IUmbracoComponentRenderer ComponentRenderer => Ensure(_componentRenderer);
+
+        [Obsolete("Inject and use an instance of " + nameof(ICultureDictionaryFactory) + " in the constructor for using it in classes or get it from Current.CultureDictionaryFactory in views.")]
         private ICultureDictionaryFactory CultureDictionaryFactory => Ensure(_cultureDictionaryFactory);
 
         /// <summary>
         /// Gets the tag context.
         /// </summary>
+        [Obsolete("Inject and use an instance of " + nameof(ITagQuery) + " in the constructor for using it in classes or get it from Current.TagQuery in views.")]
         public ITagQuery TagQuery => Ensure(_tagQuery);
 
         /// <summary>
         /// Gets the query context.
         /// </summary>
+        [Obsolete("Inject and use an instance of " + nameof(IPublishedContentQuery) + " in the constructor for using it in classes or get it from Current.PublishedContentQuery in views")]
         public IPublishedContentQuery ContentQuery => Ensure(_publishedContentQuery);
 
         /// <summary>
         /// Gets the membership helper.
         /// </summary>
+        [Obsolete("Inject and use an instance of " + nameof(Security.MembershipHelper) + " in the constructor instead")]
         public MembershipHelper MembershipHelper => Ensure(_membershipHelper);
```

Comment:
Just noted that there's not an "or get it from Current. to use it in views" option on this property, so wondering what the approach would be here if someone wanted to use `MembershipHelper` in views?  If it's planned to support that, then probably should update the Obsolete attributes description to explain how.

Reply:
The only reason not to add it to Current, is because it is available from `UmbracoViewPage`

So you should already be able to just use `@Members` in views.. I will add that to the message